stan/math/rev/mat/fun/adj_jac_apply.hpp

```diff
@@ -66,10 +352,15 @@ struct adj_jac_vari : public vari {
     Eigen::Matrix<double, Eigen::Dynamic, 1> y_adj(M_);
     for (int m = 0; m < M_; ++m)
       y_adj(m) = y_vi_[m]->adj_;
-    Eigen::Matrix<double, Eigen::Dynamic, 1> y_adj_jac
-        = f_.multiply_adjoint_jacobian(y_adj);
-    for (int n = 0; n < N_; ++n)
-      x_vi_[n]->adj_ += y_adj_jac(n);
+    auto y_adj_jacs = f_.multiply_adjoint_jacobian(is_var_, y_adj);
+
+    /**
+     * NOTE TO REVIEWER: If the cost of this lambda function redirection seems
+     * to high, we could make accumulate_adjoints a non member function and just
+     * pass it all the info it needs.
+     */
```

Comment:
The best way to figure out the cost is to test it.  I'm not sure how much overhead is involved where with the lambdas.  If they don't bind anything, they're simple, but here it's binding `this`.  I'd have thought you'd have wanted to bind it by reference, but I guess it's actually just a pointer until it's used as `*this` or `this->...`.

Reply:
I went to Google how to bind by reference, and found out you cannot bind "this" by reference: https://stackoverflow.com/questions/16323032/why-cant-i-capture-this-by-reference-this-in-lambda . Who knew?